### ISSUE TITLE
Doesn't work on macOS with big files

### ISSUE THREAD

**Issue by VolodymyrBS** [mob-sakai/CSharpCompilerSettingsForUnity#2]
Compiler fails to compile the file with error.
```
error CS1504: Source file '***/Assets/TrueSync/Physics/Farseer/Common/PhysicsLogic/RealExplosion.cs' could not be opened -- GetCPInfoExW
```

**To Reproduce**
1. Clone repo [https://github.com/VolodymyrBS/CSharpCompilerSettingsForUnityBugRepro.git](https://github.com/VolodymyrBS/CSharpCompilerSettingsForUnityBugRepro.git)
2. Import project on macOS
3. Check Console
4. See an error

**Environment:**
 - Version 1.0.1
 - Platform: Editor Mac
 - Unity version: [e.g. 2019.3.5f1]
 - Build options: any
 - Injected compiler version - 3.5.0, 3.7.0

**Additional context**
Probably it's related to this issue
[https://github.com/mono/mono/issues/12603](https://github.com/mono/mono/issues/12603)
ILSpy shows that System.Text.Encoding.CodePages.dll library inside Microsoft.Net.Compiler package use WinApi

![image](https://user-images.githubusercontent.com/20375870/94810480-bb109480-03fc-11eb-8dad-c0db06229981.png)

Note: Can't share the original project but only part of it enough to reproduce the error.
Note: Error will also occure if you try to directly call `mono <path to downloaded csc.exe> <path to RealExplosion.cs>`

**Comment by mob-sakai** [comment]
@VolodymyrBS 
Thank you for reporting.
The issue was reproduced in my environment.

Using Unity's built-in mono, the build was failed.

```
$ /Applications/Unity/Hub/Editor/2019.3.2f1/Unity.app/Contents/MonoBleedingEdge/bin/mono --version
Mono JIT compiler version 5.11.0 ((HEAD/4fbd4767c98 Mon Dec 16 02:55:50 GMT 2019)
...

$ /Applications/Unity/Hub/Editor/2019.3.2f1/Unity.app/Contents/MonoBleedingEdge/bin/mono -noconfig @rsp_test
Microsoft (R) Visual C# Compiler version 3.5.0-beta4-20153-05 (20b9af91)
Copyright (C) Microsoft Corporation. All rights reserved.

error CS1504: Source file '/Users/takashi.sakai/repos/CSharpCompilerSettingsForUnityBugRepro/Assets/TrueSync/Physics/Farseer/Common/PhysicsLogic/RealExplosion.cs' could not be opened -- GetCPInfoExW
```

Using the new version of mono, the build was successful.

```
$ mono --version
Mono JIT compiler version 6.12.0.90 (2020-02/d3daacdaa80 Thu Jul 30 16:18:06 EDT 2020)
...

$ mono -noconfig @rsp_test
Microsoft (R) Visual C# Compiler version 3.5.0-beta4-20153-05 (20b9af91)
Copyright (C) Microsoft Corporation. All rights reserved.

Assets/TrueSync/Unity/TSTransform2D.cs(87,22): warning CS0649: Field 'TSTransform2D._serialized' is never assigned to, and will always have its default value false
```


Response file: [test_rsp](https://github.com/mob-sakai/CSharpCompilerSettingsForUnity/files/5315460/test_rsp.txt)

**Comment by mob-sakai** [comment]
Plans:

- Add `Use System Mono' option
  - Pros: Simple.
  - Cons: All developers have to install mono on their machines.

- Use self-contained csc
  - Pros: Mono is not needed to run csc.
  - Cons: Increasing the download size of csc
  - Cons: csc package needs to be rebuilt.

**Comment by mob-sakai** [comment]
Fixed in 1.1.0
